api/query/atoms.go

```diff
@@ -365,6 +365,26 @@ func (t AbstractTestLabel) BindToRuns(_ ...shared.TestRun) ConcreteQuery {
 	}
 }
 
+// AbstractTestWebFeature represents the root of a web_feature query, which matches test-level
+// metadata to a searched web feature.
+type AbstractTestWebFeature struct {
+	WebFeature      string
+	manifestFetcher searchcacheWebFeaturesManifestFetcher
+}
+
+// BindToRuns for AbstractTestWebFeature fetches test-level metadata; it is independent of test runs.
+// nolint:ireturn // TODO: Fix ireturn lint error
+func (t AbstractTestWebFeature) BindToRuns(_ ...shared.TestRun) ConcreteQuery {
```

Comment:
This function needs a test in api/query/atoms_test.go, a similar setup to [TestStructuredQuery_bindTestLabel](https://github.com/web-platform-tests/wpt.fyi/blob/701c625dc22f6dedea3964de971c4c9d64191504/api/query/atoms_test.go#L1135).

Reply:
In the test setup, searchcacheWebFeaturesManifestFetcher{} will be passed in from `t.manifestFetcher`, instead of being created directly here. To make the test work, a mock file need to be generated for searchcacheWebFeaturesManifestFetcher through MockGen. 
```
if t.metadataFetcher == nil {
		t.metadataFetcher = searchcacheMetadataFetcher{}
	}
```

Reply:
In order to prevent the `if variable == nil{ then initialize}` logic (which would lead to an non-testable branch), we can instead embed AbstractTestWebFeature with an intermediate struct during the initialization of AbstractTestWebFeature. And during the initialization, we can also have a default implementation of the interface set.


You can see that happening[ here](https://github.com/web-platform-tests/wpt.fyi/pull/3665/files#diff-6b9fa200c6337e946daaabc6bc1eaba8123f839e7606201c94f28c5b61d903a8R1239-R1245). The intermediate type is called TestWebFeatureAtom. So now the UnmarshalJSON is handled by that. And if successful, it gets embedded into the AbstractTestWebFeature which is used like the other abstract atoms.

You can see in the atoms test file that now, we can override the interface with our mock still without `if variable == nil{ then initialize}`

Also, I made AbstractTestWebFeature and TestWebFeature (and as a result also TestWebFeatureAtom) all exported to be similar to what exists. Although, these and the existing ones don't seem to be needed outside of their current package.

I did not make my new interface `webFeaturesManifestFetcher` exported nor in the shared package since right now, it is only planned for the searchcache.